### PR TITLE
fix: remove vercel installation

### LINKED ISSUE #6: Support PNPM and Yarn
Right now `next-on-pages` assumes users are using NPM: https://github.com/cloudflare/next-on-pages/blob/main/src/index.ts#L41

Would be nice if this supported PNPM/Yarn as primary package managers too

### LINKED ISSUE #17: Try not to install `vercel` every time
Hello.

Are there any reasons why it trys to run `npm install -D vercel` every time?
I have already installed via `yarn add -D @cloudflare/next-on-pages vercel`.
Also, it is marked as peer-dependency which I don't think `next-on-pages` should install it for them.
Or, check `package.json` for `vercel` before installing.

Anyway I get some errors running the command.

```
Error: spawn npm ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npm',
  path: 'npm',
  spawnargs: [ 'install', '-D', 'vercel' ]
}

Node.js v18.12.0
```

node: 18.12.0
next-on-pages: 0.1.0

### PR DESCRIPTION
**Pull request by raisedadead**
The step executing the actual build defaults to `npx`. The redundant Vercel CLI installation can be removed.

This automatically means that `pnpm` is now supported without fuss (package managers are recognized correctly and automatically) and slightly cleaner.

The main point it Vercel CLI only needs the config files to be in place, and that's what the code change does.

Closes #17
Closes #6

### CODE REVIEW

**Comment by raisedadead** [comment]
Here is a sample output with pnpm, for example:

<details>
<summary>Details</summary>

```txt
pnpm build:for-edge

> portfolio@0.1.0 build:for-edge /Users/mrugesh/DEV/raisedadead/portfolio
> npx @cloudflare/next-on-pages --experimental-minify

⚡️ @cloudflare/next-on-pages CLI
⚡️
⚡️ Building project with 'npx vercel build'...
⚡️
▲ Vercel CLI 28.15.3
▲
▲ Warning: Detected "engines": { "node": ">=18" } in your `package.json` that will automatically upgrade when a new major Node.js Version is released. Learn More: http://vercel.link/node-version
▲
▲ Detected `pnpm-lock.yaml` generated by pnpm 7...
▲
▲ WARNING: You should not upload the `.next` directory.
▲
▲ Installing dependencies...
▲
▲ Lockfile is up to date, resolution step is skipped
▲
▲ Already up to date
▲
▲
▲
▲
▲ > portfolio@0.1.0 prepare /Users/mrugesh/DEV/raisedadead/portfolio
▲ > husky install
▲
▲
▲ husky - Git hooks installed
▲
▲ Done in 698ms
▲
▲ Detected Next.js version: 13.1.6
▲
▲ Running "pnpm run build"
▲
▲
▲ > portfolio@0.1.0 build /Users/mrugesh/DEV/raisedadead/portfolio
▲ > next build
▲
▲
▲ warn  - You have enabled experimental features (nextScriptWorkers, runtime) in next.config.mjs.
▲
▲ warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
▲
▲
▲
▲ info  - Linting and checking validity of types...
▲
▲ info  - Creating an optimized production build...
▲
▲ warn  - You are using the experimental Edge Runtime with `experimental.runtime`.
▲
▲ info  - Compiled successfully
▲
▲ info  - Collecting page data...
▲
▲ info  - Generating static pages (0/2)
▲
▲ info  - Generating static pages (2/2)
▲
▲ info  - Finalizing page optimization...
▲
▲
▲
▲ Route (pages)                              Size     First Load JS
▲ ┌ ℇ /                                      5.83 kB          81 kB
▲ ├   /_app                                  0 B            73.2 kB
▲ ├ ○ /404                                   181 B          73.3 kB
▲ ├ ℇ /api/ping                              0 B            73.2 kB
▲ ├ ℇ /privacy                               8.42 kB        83.6 kB
▲ ├ ℇ /refunds                               2.39 kB        77.6 kB
▲ └ ℇ /terms                                 8.67 kB        83.9 kB
▲ + First Load JS shared by all              78.9 kB
▲   ├ chunks/framework-ac88a2a245aea9ab.js   45.2 kB
▲   ├ chunks/main-c760948dd93bcd41.js        26.8 kB
▲   ├ chunks/pages/_app-56e2c3b1fb7093fc.js  293 B
▲   ├ chunks/webpack-ee7e63bc15b31913.js     815 B
▲   └ css/912105f72fed41d3.css               5.71 kB
▲
▲
▲ ℇ  (Streaming)  server-side renders with streaming (uses React 18 SSR streaming or Server Components)
▲ ○  (Static)     automatically rendered as static HTML (uses no initial props)
▲
▲
▲ Collected static files (public/, static/, .next/static): 2.816ms
▲
▲ Build Completed in .vercel/output [5s]
▲
⚡️
⚡️
⚡️ Completed 'npx vercel build'.
⚡️
⚡️ Generated '.vercel/output/static/_worker.js'.
```

</details>

where the `pnpm build:for-edge` is just a run-script in my package.json that does the same as below:

```
npx @cloudflare/next-on-pages --experimental-minify
```

**Comment by sgoodluck** [comment]
This would be excellent!

**Comment by sgoodluck** [comment]
Did this work for you when deploying to cloudflare pages as well?

**Comment by raisedadead** [comment]
I am not using CF deployments; I am on GitHub actions for unrelated reasons (they do not support Node 18). But I can set up a demo project if you need me to validate this.

**Comment by sgoodluck** [comment]
I would greatly appreciate that. 

I can get this working when I set the entire project to experimental edge BUT every page (including simple should be static only ones) are streaming.

As soon as the build produces static pages, the deployment leads to 404s.

**Comment by raisedadead** [comment]
Sorry – I should have been clear. 

I am not an expert on why you face streaming vs. static content issues. You should probably report your use case as an issue to the maintainers.

I can speak to the context of this PR only, which is it removes the redundant installation of the CLI into dev-dependencies.

That fixes the build workflow and in theory should not change anything with the CF-based build pipelines.

In short what you are asking is something which I can't test sadly :)

**Comment by sgoodluck** [comment]
Ah, well, I appreciate you taking a glance anyway! 

I did a clean setup and build using the standard setup and another pulling in your commit using pnpm. Both worked! 

Weirdly, the standard NPM setup was slightly faster maybe due to the pnpm install step. Locally, working with pnpm is much better though!

Screen Shot 2023-02-14 at 11 58 01 AM

**Comment by raisedadead** [comment]
Yup – `pnpm` is just super nice. 

If you are curious about my project [here is my config](https://github.com/raisedadead/portfolio/blob/d5871118007afc19f3d73b6965980686c052d788/.github/workflows/build-and-deploy.yml#L63-L121)

With a note that my build step is `npx @cloudflare/next-on-pages --experimental-minify` – mind the use of `npx` and not `pnpx`.

This is deliberate because if you have `pnpm` as your package manager, then Vercel CLI will pick it up automatically. So you only added time would be in the initial `npm install -g pnpm` – but that's about it.

In my setup, this PR is already being used by another nifty feature of `pnpm` which patches the change into the `@cloudflare/next-on-pages` [as seen here](https://github.com/raisedadead/portfolio/blob/d5871118007afc19f3d73b6965980686c052d788/patches/%40cloudflare__next-on-pages%400.3.0.patch).

Hope this helps you in someway. Thanks for your feedback and comments

**Comment by sgoodluck** [comment]
Interestingly - **when deploying via Cloudflare and not via GitHub actions**:

Fails: `npm i -g pnpm && npx @cloudflare/next-on-pages --experimental-minify`

Succeeds: `npm i -g pnpm && pnpx @cloudflare/next-on-pages --experimental-minify`

This may have to do with it not being built into cloudflare pages directly: 
https://community.cloudflare.com/t/add-pnpm-to-pre-installed-cloudflare-pages-tools/288514/6

**Comment by dario-piotrowicz** [comment]
@raisedadead thanks so much for the PR! 😃 

sorry it took us a while to check it out 🙏 

It looks very good, we just want to do some last double checking and it should be good to go 😄 

but just one last thing, could you please create a changeset? (via `npx changeset`)

**Comment by raisedadead** [comment]
No worries @dario-piotrowicz!

Thanks a lot for taking the time. I have updated the PR as you requested (using changeset for the first-time, please let me know if I messed up).

Thanks.

**Comment by dario-piotrowicz** [comment]
@raisedadead awesome, the changeset looks great! thanks a lot 😄

**Review by GregBrimble — APPROVED** [review]
Thank you for your PR!
